presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -593,6 +594,45 @@ public static Boolean stWithin(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType
         return leftGeometry.within(rightGeometry);
     }
 
+    @ScalarFunction
+    @Description("Calculates the great-circle distance between two points on the Earth's surface")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double greatCircleDistance(@SqlType(StandardTypes.DOUBLE) double latitude1, @SqlType(StandardTypes.DOUBLE) double longitude1, @SqlType(StandardTypes.DOUBLE) double latitude2, @SqlType(StandardTypes.DOUBLE) double longitude2)
+    {
+        checkLatitude(latitude1);
+        checkLongitude(longitude1);
+        checkLatitude(latitude2);
+        checkLongitude(longitude2);
+
+        latitude1 = Math.toRadians(latitude1);
+        longitude1 = Math.toRadians(longitude1);
+        latitude2 = Math.toRadians(latitude2);
+        longitude2 = Math.toRadians(longitude2);
+
+        double deltaLongitude = longitude1 - longitude2;
+
+        double t1 = Math.cos(latitude2) * Math.sin(deltaLongitude);
+        double t2 = Math.cos(latitude1) * Math.sin(latitude2) - Math.sin(latitude1) * Math.cos(latitude2) * Math.cos(deltaLongitude);
+        double t3 = Math.sin(latitude1) * Math.sin(latitude2) + Math.cos(latitude1) * Math.cos(latitude2) * Math.cos(deltaLongitude);
+        double angularDistance = Math.atan2(Math.sqrt(t1 * t1 + t2 * t2), t3);
+
+        return angularDistance * EARTH_RADIUS_KM;
+    }
+
+    private static void checkLatitude(double latitude)
```

Comment:
You can merge these to functions into `checkCoordinate(double latitude, double longitude)`.

Reply:
@nezihyigitbasi term `coordinate` is quite generic; in this file some functions operate on arbitrary planes with (x, y) coordinates not limited and other functions work with lat/lon points which are restricted. It seems misleading to have `checkCoordinate` function that assumes lat/lon.